### PR TITLE
images: Switch to Fedora Rawhide FreeIPA container for the "services" image

### PR DESCRIPTION
**Pull request by mvollmer**
This is the best alternative to the :latest tag, which has been dropped.

 * [x] image-refresh services

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-095858/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-095858/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-112323/

File: images/scripts/services.setup [hunk 1]
```diff
@@ -32,7 +34,7 @@ docker run -d --rm --name freeipa -ti -h f0.cockpit.lan \
     -e IPA_SERVER_IP=$SERVER_IP \
     -p 53:53/udp -p 53:53 -p 80:80 -p 443:443 -p 389:389 -p 636:636 -p 88:88 -p 464:464 -p 88:88/udp -p 464:464/udp -p 123:123/udp \
     -v /var/lib/ipa-data:/data:Z \
-    freeipa/freeipa-server \
+    freeipa/freeipa-server:fedora-raw-hide \
```
Comment (martinpitt) [inline images/scripts/services.setup:37]: The extra dash needs to go. :grin: Please do `./tests-trigger 1238 image:services` after pushing, to trigger a set of representative tests that use that image.

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-112323/

File: images/scripts/services.setup (hunk 1 above)
Comment (mvollmer) [reply images/scripts/services.setup:37]: > The extra dash needs to go. 

Pilot error! (Ouch)

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-114747/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-114747/

**Comment by mvollmer** [comment]
```
+ /run-freeipa
02eb495aeccb7e6926284361a16ccb6df734eaef0ff3816d25fd76acfee18707
/usr/bin/docker-current: Error response from daemon: oci runtime error: container_linux.go:235: starting container process caused "process_linux.go:339: running prestart hook 2 caused \"error running hook: exit status 1, stdout: , stderr: \"".
```

Let's try centos-8 just to see what happens.

**Comment by cockpituous** [comment]
image-refresh in progress on 3-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-120404/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1238-20200917-120404/

**Comment by mvollmer** [comment]
> Let's try centos-8 just to see what happens.

Same thing, I debug this locally.

**Comment by mvollmer** [comment]
From the journal:

```
Sep 18 03:26:46 m1.cockpit.lan oci-systemd-hook[1831]: systemdhook <error>: 1aec67092ac5: Failed to open /var/lib/containers/storage/overlay/63b4cf9b4a5df56e1136ade945306906dc00b41b5a486c1bc5614c03e43c6516/merged/etc/machine-id for writing: No such file or directory
```

This hook is only used for containers that use systemd as their entry point, and freeipa-server is one of those. I guess they are rare, and 'normal' containers that don't use systemd inside themselves, such as "httpd", work just fine.

The hook error happens on our centos-7 image with both docker and podman. Our fedora-32 image can't run docker anymore bc of cgroupsv2, but podman can start the freeipa-container successfully. So we might switch to fedora-32 and podman. I'll try a couple more combinations...

**Comment by mvollmer** [comment]
With our centos-8-stream image and podman:

```
# podman run --name freeipa-server-container -ti     -h ipa.example.test -v /var/lib/ipa-data:/data:Z freeipa/freeipa-server:fedora-32
systemd v245.7-1.fc32 running in system mode. (+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified)
Detected virtualization container-other.
Detected architecture x86-64.
Set hostname to <ipa.example.test>.
Failed to create /machine.slice/libpod-17c7a2e977acba5004315df501700f26553f2f10f291c225421966d7374bdf92.scope/init.scope control group: Read-only file system
Failed to allocate manager object: Read-only file system
[!!!!!!] Failed to allocate manager object.
Exiting PID 1...
```

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-02.
Log: https://logs.cockpit-project.org/logs/image-refresh-1238-20200918-095050/

**Comment by mvollmer** [comment]
Let's try with a couple of hacks that have been suggested in https://github.com/freeipa/freeipa-container/issues/348

**Comment by cockpituous** [comment]
image-refresh services done: https://github.com/cockpituous/bots/commits/image-refresh-services-20200918-103131

**Comment by mvollmer** [comment]
Cool, we have an image!

**Comment by mvollmer** [comment]
I think testNodeNavigation (check_openshift.TestOpenshift) on rhel-7-9@cockpit-project/cockpit/rhel-7.9 is just broken. It certainly seems unrelated.

File: images/scripts/services.setup [hunk 2]
```diff
@@ -22,6 +22,9 @@ systemctl enable --now docker
 #
 #############
 
+# see https://github.com/freeipa/freeipa-container/issues/348
+rm /usr/libexec/oci/hooks.d/oci-systemd-hook
```
Comment (martinpitt) [inline images/scripts/services.setup:26]: Seems a bit cleaner to uninstall the package, but *shrug*, good enough.

**Review by martinpitt — APPROVED** [review]
Thanks! testNodeNavigation has been super-flaky forever, so please ignore that. This still needs a Closes # or squashing, but looks fine otherwise. Thanks!

File: images/scripts/services.setup (hunk 2 above)
Comment (mvollmer) [reply images/scripts/services.setup:26]: Uninstalling the package takes all of docker with it.

**Review by martinpitt — APPROVED** [review]
Danke!

**Pull request merged**